src/libraries/rest/router/rest_router.cpp

```
@@ -0,0 +1,134 @@
+            };
+            size_t start = 0, end = 0;
+            while ((end = query.find('&', start)) != std::string::npos)
+            {
+                handle_param(start, end - start);
+                start = end + 1;
+            }
+
+            // Handle the last parameter
+            handle_param(start, query.length() - start);
+
+            url = url.substr(0, query_start);
```

cppcheck (2.10-2)

[performance] Ineffective call of function 'substr' because a prefix of the string is assigned to itself. Use resize() or pop_back() instead.

(uselessCallsSubstr)